pkg/discovery/discovery.go

```diff
@@ -110,15 +110,21 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 	}
 
 	var pdClients []pdapi.PDClient
+
+	if tc.Spec.PD != nil {
+		pdClients = append(pdClients, d.pdControl.GetPDClient(pdapi.Namespace(tc.GetNamespace()), tc.GetName(), tc.IsTLSClusterEnabled()))
+	}
+
 	if tc.Spec.Cluster != nil && len(tc.Spec.Cluster.Name) > 0 {
 		namespace := tc.Spec.Cluster.Namespace
 		if len(namespace) == 0 {
 			namespace = tc.GetNamespace()
 		}
 		pdClients = append(pdClients, d.pdControl.GetClusterRefPDClient(pdapi.Namespace(namespace), tc.Spec.Cluster.Name, tc.Spec.Cluster.ClusterDomain, tc.IsTLSClusterEnabled()))
 	}
-	if tc.Spec.PD != nil {
-		pdClients = append(pdClients, d.pdControl.GetPDClient(pdapi.Namespace(tc.GetNamespace()), tc.GetName(), tc.IsTLSClusterEnabled()))
+
+	for _, pdMember := range tc.Status.PD.PeerMembers {
+		pdClients = append(pdClients, d.pdControl.GetPeerPDClient(pdapi.Namespace(ns), tc.Name, tc.IsTLSClusterEnabled(), pdMember.ClientURL, pdMember.Name))
```

Comment:
Is there any potential problem for the `if member.Name == podName` bellow if multiple members from different clusters have the same pod name (may also from `tc.Spec.Cluster`?).

Reply:
Oh, it's a missing change for multi-cluster, I will submit another PR for this. Thanks a lot. 

For now, this check won't work. The PD member name in multi-cluster will use the FQDN(cluster2-pd-0.cluster2-pd-peer.pingcap.svc.cluster2.local), rather than pod name(cluster2-pd-0).

Reply:
@handlerww @csuzhangxc The change is done in https://github.com/pingcap/tidb-operator/pull/3365/, I had discussed that with @lichunzhu, it seems that it's not necessary for PD, please confirm with him and if no need for PD, we can remove the check in another PR.

Reply:
I suggest keeping this for safety. PD will return an error if `cfg.Join == cfg.AdvertiseClientUrls`. https://github.com/tikv/pd/blob/bb5abcf0b9b38c845b6afc7d55d7317ef23363d4/server/join/join.go#L87
However, PD will use its own cached data when restarting.